coding.go

```diff
@@ -34,64 +34,63 @@ func (nb _PBNode__NodeBuilder) DecodeDagProto(r io.Reader) error {
 	if err := pbn.Unmarshal(encoded); err != nil {
 		return fmt.Errorf("unmarshal failed. %v", err)
 	}
-	pbLinks := make([]PBLink, 0, len(pbn.Links))
-	for _, link := range pbn.Links {
-		hash, err := cid.Cast(link.GetHash())
+	return fluent.Recover(func() {
+		fb := fluent.WrapAssembler(nb)
+		fb.CreateMap(0, func(fmb fluent.MapAssembler) {
+			fmb.AssembleEntry("Links").CreateList(len(pbn.Links), func(flb fluent.ListAssembler) {
+				for _, link := range pbn.Links {
+					hash, err := cid.Cast(link.GetHash())
 
-		if err != nil {
-			return fmt.Errorf("unmarshal failed. %v", err)
-		}
-		pbLinks = append(pbLinks, PBLink{
-			d: PBLink__Content{
-				Hash: MaybeLink{
-					Maybe: schema.Maybe_Value,
-					Value: Link{cidlink.Link{Cid: hash}},
-				},
-				Name: MaybeString{
-					Maybe: schema.Maybe_Value,
-					Value: String{link.GetName()},
-				},
-				Tsize: MaybeInt{
-					Maybe: schema.Maybe_Value,
-					Value: Int{int(link.GetTsize())},
-				},
-			},
+					if err != nil {
+						panic(fluent.Error{Err: fmt.Errorf("unmarshal failed. %v", err)})
+					}
+					flb.AssembleValue().CreateMap(0, func(fmb fluent.MapAssembler) {
+						fmb.AssembleEntry("Hash").AssignLink(cidlink.Link{Cid: hash})
+						fmb.AssembleEntry("Name").AssignString(link.GetName())
+						fmb.AssembleEntry("Tsize").AssignInt(int(link.GetTsize()))
```

Comment:
Same question as above about name strings.  Does this panic if the field isn't in the protobuf?  Or silently coerce it to a zero?